controllers/hetznerbaremetalhost_controller.go

```diff
@@ -170,6 +170,14 @@ func (r *HetznerBareMetalHostReconciler) reconcileSelectedStates(ctx context.Con
 	case infrav1.StateNone:
 		var needsUpdate bool
 		if !bmHost.DeletionTimestamp.IsZero() && bmHost.Spec.ConsumerRef == nil {
+			conditions.MarkFalse(
```

Comment:
you should set this condition independent of Spec.ConsumerRef. So just split up the if statements to set the condition always when the deletion timestamp is not nil